examples/with-react-intl/components/PageWithIntl.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react'
+import {IntlProvider, addLocaleData, injectIntl} from 'react-intl'
+
+// Register React Intl's locale data for the user's locale in the browser. This
+// locale data was added to the page by `pages/_document.js`. This only happens
+// once, on initial page load in the browser.
+if (typeof window !== 'undefined' && window.ReactIntlLocaleData) {
```

Comment:
There are several issues with Next at play here:

1. `page/_document.js` is bundled with Webpack, and errors our if you try to use `fs` or any other built-in Node modules even though it only runs on the client.

2. The `__NEXT_DATA__` is serialized using `JSON.stringify()`. This means that `getInitialProps()` must return an object that's JSON-serializable, and React Intl's locale data isn't JSON-serializable since it contains functions for pluralization.

3. Calling `addLocaleData()` on every page transition wouldn't be ideal. It's best to only call this function one time.

It's possible to address 1 and 2 of these issues in Next, and I'd be happy to rework example once they are changed, but as it stands now there isn't a good way around it.

For 2, [`serialize-javascript`](https://github.com/yahoo/serialize-javascript) can be used which doesn't support serializing functions. Changing Next to use this instead of raw `JSON.stringify()` would remove the need for this example to use a custom `pages/_document.js`.